streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -608,6 +618,11 @@ public void checkpoint() {
         // checkpoint those stores that are only logged and persistent to the checkpoint file
         final Map<TopicPartition, Long> checkpointingOffsets = new HashMap<>();
         for (final StateStoreMetadata storeMetadata : stores.values()) {
+            // do not checkpont non-transactional state stores under EOS
+            if (eosEnabled && !storeMetadata.stateStore.transactional()) {
```

Comment:
As we discussed in the KIP, I think we can just remove this branched logic as well and always checkpoint.